src/portable/synopsys/dwc2/dcd_dwc2.c

```diff
@@ -47,7 +47,37 @@
 // MACRO TYPEDEF CONSTANT ENUM
 //--------------------------------------------------------------------+
 
+#ifdef DWC2_ENABLE_MEM_CACHE
+
+#ifndef DWC2_MEM_CACHE_LINE_SIZE
+#warning "Cache line size not specified, use default(64b) instead"
+#    define DWC2_MEM_CACHE_LINE_SIZE 0x40
+#endif // DWC2_MEM_CACHE_LINE_SIZE
+
+CFG_TUD_MEM_SECTION struct {
+  union {
```

Comment:
This is a manual tweak to protect memory after the aligned `_setup_packet` buffer. 
I have doubts, that ` __attribute__((aligned(x))` protects the memory after non-aligned variable, so I decided to solve it this way. At least there is nothing about that in the GCC doc.
Anyway, suggestions are welcome. 

I referred to: https://gcc.gnu.org/onlinedocs/gcc/Common-Variable-Attributes.html#index-aligned-variable-attribute

For class buffers (such as `mscd cbw` and `csw`, `cdc epout` and `epin` and so on, it is better to optimize in class driver code and different PR, IMHO). But anyway also should be done.

Reply:
Yeah, this will be a bit of an headache. gcc attribute alignment only ensure the variale alignement only, its size can be smaler, that would leave rom for other variable/data which can be corrupted when memroy to cache is called. I will try to figure out a way to address this.